### ISSUE TITLE
[Trigger Email] The functions aren't working after update

### ISSUE THREAD

**Issue by mauzepeda** [firebase/extensions#160]
If I go to Function Details in GCP the function displays an error sign with the message "Unknown function status". If I go to the link in the source location the window automatically closes, if I try to download the zip file with the source, it says "NoSuchBucket", ''The specified bucket does not exist.'.
It seems like the source is not being downloaded from cloud storage.

### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs for a **specific extension in this repository**.
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

- For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
  with the firebase tag.
- For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
  google group.
- To file a bug against the Firebase Extensions platform, or for an issue affecting multiple extensions, please reach out to
  [Firebase support](https://firebase.google.com/support/troubleshooter/contact/) directly.

### [REQUIRED] Step 2: Describe your configuration

- Extension name: **\_** (`storage-resize-images`, `firestore-send-email`, etc)
- Extension version: **\_**
- Configuration values (redact info where appropriate):
  - **\_**
  - **\_**

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, [log/console output](https://firebase.google.com/docs/extensions/manage-installed-extensions#view-logs), etc.

**Comment by Ehesp** [comment]
Hey @mauzepeda

Sorry to hear you're having issues. 

XFN Team: Are there any logs that can be provided here to help debug the issue - not come across this one before.

**Comment by mauzepeda** [comment]
It seems like the issue is related to the fact that removing and
reinstalling an extension is not equivalent to installing an extension for
the first time.
If I delete the function, and reinstall it, the Stackdriver logs show as if
it is the same function and it was never deleted. Is this expected? I also
suspect
it could be related to permissions / service accounts (and the deletion and
reinstall of these). The specific error I get is: "Deployment failure:
failed to retrieve function source code.".
If I go to the "source" tab and I click on the link under source location
then the file downloads, but if I click in the button to download the
source, I get a "NoSuchBucket" error message.
The logs show 3 type of Errors:

   - CreateFuncion INVALID_ARGUMENT
   - (followed by) CreateFuncion ALREADY_EXISTS
   - UpdateFuncion INVALID_ARGUMENT

I hope this helps!

**Comment by laurenzlong** [comment]
Thanks for the repro instructions! Let's check your hypothesis that deleting and reinstalling the same extension is what causes the issue. Can you try to install another instance of the same extension? (Without uninstalling the failed extension, go to https://firebase.google.com/products/extensions/ and install the same extension again. It will ask you to pick a different instance name, pick whatever you want). Let me know if this works or fails.

**Comment by mauzepeda** [comment]
Hi @laurenzlong , thank you for the follow up! This issue was being caused because I accidentally deleted the bucket where the functions lived (don't ask me how I did this). I worked with the firebase support team to restore this bucket because I was unable to deploy functions from firebase-tools too. After the bucket was restored and I reinstalled TriggerMail, the extension was able to write the functions on the bucket and work as expected. I tried with a difference instance name and that worked too.
